libs/@local/graph/postgres-store/src/permissions/mod.rs

```diff
@@ -862,108 +925,448 @@ impl<C: AsClient, A: AuthorizationApi> PostgresStore<C, A> {
+                    &principal_id.map(PrincipalId::into_uuid),
+                    &principal_type,
+                    &actor_type,
+                    &policy.resource.as_ref().map(Json),
+                ],
+            )
+            .await
+            .map_err(|error| {
+                let policy_error = match (error.code(), principal_id) {
+                    (Some(&SqlState::UNIQUE_VIOLATION), _) => {
+                        PolicyError::PolicyAlreadyExists { id: policy_id }
+                    }
+                    (Some(&SqlState::FOREIGN_KEY_VIOLATION), Some(principal_id)) => {
+                        PolicyError::PrincipalNotFound { id: principal_id }
+                    }
+                    _ => PolicyError::StoreError,
+                };
+                Report::new(error).change_context(policy_error)
+            })?;
 
-                -- All parent teams of subteams (recursively through hierarchy)
-                SELECT parent.id, parent.principal_type
-                FROM actor_role
-                JOIN role ON actor_role.role_id = role.id
-                JOIN team_hierarchy ON team_hierarchy.child_id = role.team_id
-                JOIN team parent ON parent.id = team_hierarchy.parent_id
-                WHERE actor_role.actor_id = $1
+        for action in policy.actions {
+            transaction
+                .execute(
+                    "INSERT INTO policy_action (policy_id, action_name) VALUES ($1, $2)",
+                    &[&policy_id, &action],
+                )
+                .await
+                .map_err(|error| {
+                    let policy_error = match error.code() {
+                        Some(&SqlState::FOREIGN_KEY_VIOLATION) => {
+                            PolicyError::ActionNotFound { id: action }
+                        }
+                        _ => PolicyError::StoreError,
+                    };
+                    Report::new(error).change_context(policy_error)
+                })?;
+        }
+        transaction
+            .commit()
+            .await
+            .change_context(PolicyError::StoreError)?;
+
+        Ok(policy_id)
+    }
+
+    /// Gets all policies associated with an actor.
+    ///
+    /// This provides a complete set of policies that apply to an actor, including all policies that
+    ///   - apply to the actor itself,
+    ///   - apply to the actor's roles,
+    ///   - apply to the actor's teams, and
+    ///   - apply to the actor's parent teams (for subteams).
+    ///
+    /// # Errors
+    ///
+    /// - [`PrincipalNotFound`] if the actor with the given ID doesn't exist
+    /// - [`StoreError`] if a database error occurs
+    ///
+    /// [`PrincipalNotFound`]: PrincipalError::PrincipalNotFound
+    /// [`StoreError`]: PrincipalError::StoreError
+    // TODO: We should probably allow filtering based on the action as well here
+    #[expect(clippy::too_many_lines)]
+    pub async fn get_policies_for_actor(
```

Comment:
@hashdotai this is the hearth of this module. Carefully review the function and make sure the documentation is accurate, all code paths are covered (which is hard to measure with Codecov as this largely comes from an SQL statement), and it's properly tested. (Note, that the mentioned TODO on filtering based on an action is a potential future optimization and is intentionally not considered in this PR)

Reply:
@hashdotai Could you elaborate on that?

> The handling of policy constraints is currently simple (always None)

Do you mean the tests or in this implementation? What do you mean by "policy constraints" in the first place, we have three constraints, principals, actions, and resources.

Reply:
Ah, got it. Yes, that's a placeholder for potential future constraints, but I think I'll just remove that in a follow-up.